### PR TITLE
[feature](function) support ip function ipv4stringtonum(ordefault, ornull), inet_aton

### PR DESCRIPTION
**Pull request by superdiaodiao**
## Proposed changes

**1. Issue Number: #20584**

image

**2. Issue Number: #21489**

image

**3. The Doris outcome:**

image

image

image

image

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run p0

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

File: gensrc/script/doris_builtins_functions.py [hunk 1]
```diff
@@ -2004,6 +2004,12 @@
         [['ipv4numtostring','inet_ntoa'], 'VARCHAR', ['SMALLINT'], 'ALWAYS_NULLABLE'],
         [['ipv4numtostring','inet_ntoa'], 'VARCHAR', ['INT'], 'ALWAYS_NULLABLE'],
         [['ipv4numtostring','inet_ntoa'], 'VARCHAR', ['BIGINT'], 'ALWAYS_NULLABLE'],
+        [['ipv4stringtonum','inet_aton'], 'BIGINT', ['VARCHAR'], 'ALWAYS_NOT_NULLABLE'],
```
Comment (morningman) [inline gensrc/script/doris_builtins_functions.py:2007]: These function should be implements in Nereids(the new query planner)

Comment (superdiaodiao) [reply gensrc/script/doris_builtins_functions.py:2007]: yes, I have a plan to make it the next PR

Comment (superdiaodiao) [reply gensrc/script/doris_builtins_functions.py:2007]: OK, I will add.

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
@morningman Hi, there. I have added these func for the Nereids already. Please take a code review. Thanks~~~

**Comment by superdiaodiao** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by zy-kkk — APPROVED** [review]
LGTM

**Pull request merged**